### ISSUE TITLE
`hide`, `alias`, and `def` should not allow built-ins/keywords as arguments

### ISSUE THREAD

**Issue by NotTheDr01ds** [nushell/nushell#11686]
### Describe the bug

It's possible to remove or override the definition of built-ins, with no way to recover other than restarting the shell.  This footgun simply shouldn't be allowed.

### *Benign Case*

```Nushell
> hide help
> help hide
Error: nu::shell::external_command

  × External command failed
   ╭─[entry #2:1:1]
 1 │ help hide
   · ──┬─
   ·   ╰── can't run executable
   ╰────
  help: Permission denied (os error 13)
```

As far as I know, there's no way to recover other than to exit the shell and restart.

### *Much worse:  Makes the shell unusable*

And that's the "fairly benign" case.  Hiding a built-in that is used in  a display hook will break the shell completely:

```Nushell
> hide if
Error: nu::shell::external_command

  × External command failed
   ╭─[display_output hook:1:1]
 1 │ if (term size).columns >= 100 { table -e } else { table }
   · ─┬
   ·  ╰── did you mean 'dfr arg-where'?
   ╰────
  help: No such file or directory (os error 2)
> ls
Error: nu::shell::external_command

  × External command failed
   ╭─[display_output hook:1:1]
 1 │ if (term size).columns >= 100 { table -e } else { table }
   · ─┬
   ·  ╰── did you mean 'dfr arg-where'?
   ╰────
  help: No such file or directory (os error 2)
```

Of course, the same is possible with `alias` or `def`:

```Nushell
> def if [] {}
# Similar result (but not identical) as above
> alias if = echo
╭───┬────────────────╮
│ 0 │            242 │
│ 1 │ >=             │
│ 2 │            100 │
│ 3 │ <Closure 1986> │
│ 4 │ else           │
│ 5 │ <Closure 1987> │
╰───┴────────────────╯
> ls
╭───┬────────────────╮
│ 0 │            242 │
│ 1 │ >=             │
│ 2 │            100 │
│ 3 │ <Closure 1989> │
│ 4 │ else           │
│ 5 │ <Closure 1990> │
╰───┴────────────────╯
```

### Expected behavior

I expect Nu to reject any builtin as an argument to `hide`, `def`, `alias`, or any other command that can override its definition.

The Fish shell properly handles this:

```Bash
> alias if echo
- (line 1): function: if: cannot use reserved keyword as function name
function if --wraps echo --description 'alias if echo';  echo $argv
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
from sourcing file -
        called on line 70 of file /nix/store/5lhlp2sbybr29xjbbkghxr66kc5pbz4g-fish-3.6.4/share/fish/functions/alias.fish
in function 'alias' with arguments 'if echo'
```

### Configuration

| key                | value                                                                           |
| ------------------ | ------------------------------------------------------------------------------- |
| version            | 0.88.1                                                                          |
| branch             |                                                                                 |
| commit_hash        |                                                                                 |
| build_os           | linux-x86_64                                                                    |
| build_target       | x86_64-unknown-linux-gnu                                                        |
| rust_version       | rustc 1.74.0 (79e9716c9 2023-11-13) (built from a source tarball)               |
| cargo_version      | cargo 1.74.0                                                                    |
| build_time         | 1980-01-01 00:00:00 +00:00                                                      |
| build_rust_channel | release                                                                         |
| allocator          | mimalloc                                                                        |
| features           | dataframe, default, extra, sqlite, trash, which, zip                            |
| installed_plugins  | from eml, from ics, from ini, from vcf, query, query json, query web, query xml |

**Comment by NotTheDr01ds** [comment]
I get that users coming from another shell often want something like:

```Nushell
alias ls = ^exa
```

But even that's a footgun in Nu since so many scripts are going to assume and depend on `ls` returning structured data.

That means that, even though Nu has *many* more built-ins than Fish, I think the safest path here is to simply not allow redefinition of any built-in.

**Comment by NotTheDr01ds** [comment]
Side-side note:

The [Modules -> Hiding](https://www.nushell.sh/book/modules.html#hiding) section of the Book says:

> Any custom command or alias, imported from a module or not, can be "hidden", restoring the previous definition.

This doesn't seem to be the case.  If so, a `def if [] {}` should be "fixable" with a `hide if`, right?

Is *"restoring the previous definition"* really the intended behavior of `hide`?  Or do I need to look at a Book PR on that?

**Comment by fdncred** [comment]
> It's possible to remove or override the definition of built-ins, with no way to recover other than restarting the shell. This footgun simply shouldn't be allowed.

I agree that there should be a way to recover, kind of like we use `^` for externals, there could be some sigil for internals. But I'm not sure I agree that ti shouldn't be allowed. Maybe you're right? It's worth discussing more.

**Comment by 1Kinoti** [comment]
Or, there should be a diagnostic that warns you of this and urges you to give the built-in another name first.

```nushell
> alias ls = rm * -rfp

Warning: nu::shell::aliasing_buitins

  × Aliasing builtins

   ╭─[entry #69:6:1]
 1 │ alias ls = rm * -rfp
   ·       ─┬
   ·        ╰── builtin `ls` aliased here
   ╰────
  help: Aliasing builtins can be potentially destructive.
  Consider giving the builtin `ls` another alias first
```

In a terminal, this can be interactive maybe?

I think this shouldn't be disallowed altogether because there are situations where you might need to alias some builtins. But aliasing keywords should be disallowed
